scripts/sponsors.cr

```diff
@@ -13,6 +13,17 @@ record Sponsor, name : String, url : String?, logo : String?, last_payment : Flo
   @[JSON::Field(converter: Time::Format.new("%b %-d, %Y"))]
   property since : Time
   property overrides : String?
+
+  # Merge this sponsor with the given sponsor. Requires that the name and url match.
+  def merge(other : Sponsor?) : Sponsor
+    return self if other.nil?
+    raise "Can't merge #{self} with #{other}" if name != other.name || url != other.url
+    copy_with(
+      last_payment: since > other.since ? last_payment : other.last_payment,
+      all_time: all_time + other.all_time,
+      since: since < other.since ? since : other.since,
```

Comment:
```suggestion
      last_payment: Math.max(last_payment, other.last_payment),
      all_time: all_time + other.all_time,
      since: Math.min(since : other.since),
```

Reply:
Happy to add an import to math and change `since`, but changing `last_payment` in that way changes the meaning. For example, if I contributed $100 on bountysource last in 2019, but only $5 on opencollective today, it'd take the bountysource value. Currently, I have it selecting the most recent

Reply:
Oops, sorry. Scrap the `last_payment` change.

You don't need to import `Math`, it's already in core lib.